metabase/metabase
# Performance: partial data-permissions graph refactor

Closes #44768

### Description

Updates to the BE (code already approved) and FE changes to allow for partial editing of permissions graphs in the admin settings. This change detects which groups have had some kind of modification within the graph and only sends that part of the graph / omits unmodified groups in the update request body. The backend now only sends a response with just the modified groups, which the FE will now merge back into the client's graph.

### How to verify

The changes _should_ not change the user experience in any way than increasing performance. Please verify that modifying permissions continue to do what you'd expect.

### Checklist
